include/deal.II/numerics/vector_tools_hermite.h

```diff
@@ -0,0 +1,133 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef vector_tools_hermite_h
+#define vector_tools_hermite_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/quadrature.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/types.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <map>
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace VectorTools
+{
+  /**
+   * @name Domain and boundary projection for Hermite elements
+   * @{
+   */
+
+  /**
+   * @p Enumeration representing the type of boundary condition to be enforced.
+   * Since Hermite finite elements can strongly impose continuity in various
+   * derivatives as well as value at element boundaries, they can also be used
+   * to impose boundary conditions such as Neumann conditions directly to reduce
+   * the size of the computational system.
+   */
+  enum HermiteBoundaryType
+  {
+    hermite_dirichlet,
+    hermite_neumann,
+    hermite_2nd_derivative,
+  };
```

Comment:
I am wondering if this general enough. Would someone to fix multiple components? What about `hermite_5th_derivative`? Wouldn't an approach via `ComponentMask` more flexible?

Reply:
?